web/src/stream_edit.js

```diff
@@ -319,6 +319,13 @@ function stream_setting_changed(e) {
         sub[setting] =
             user_settings[settings_config.generalize_stream_notification_setting[setting]];
     }
+    if (setting === "pin_to_top") {
+        $status_element = $(`#stream_change_property_status${CSS.escape(sub.stream_id)}`);
+    } else {
+        $status_element = $(
+            `#stream_change_notification_setting_status${CSS.escape(sub.stream_id)}`,
+        );
```

Comment:
It would be better if just not use this `if` condition and instead find the element using `.closest(".alert-notification")`. This way the status element would not need ID attribute as well.

Reply:
Thanks @sahil839  I will implement the PR  using  ```.closest(".alert-notification").```
Before  that  I have found that   ```stream_is_muted_changed()```  uses  repeated logic .
so  I have  updated the PR removing that function and  handled the  ```is_muted``` property in ```stream_setting_changed()``` everything is working fine in terms of output. please tell me , am I missing out  something?Is this correct?.

Reply:
I think that would work fine by reading the code but I have not tested it yet. Can do it once the PR is updated with all the required changes. But that change should be in its own separate commit with proper explaination in the commit message and not in the commit for fixing the indicator.